File: Techschool/Techschool.BLL/Models/Vacations/OtherVacationModel.cs
using System;

namespace Techschool.BLL.Models.Vacations
{
    public class OtherVacationModel : VacationModel
    {
        public string TypeOfVacation { get; set; }
        public string OrderNumber { get; set; }
        public DateTime OrderDate { get; set; }
        public string Notes { get; set; }
    }
}

File: Techschool/Techschool/ClientApp/src/app/models/vacations/other-vacation.model.ts
import { VacationModel } from './vacation.model';

export class OtherVacationModel extends VacationModel {
  typeOfVacation: string;
  orderNumber: string;
  orderDate: Date;
  notes: string;

  constructor (data: OtherVacationModel = null) {
    super();
    if (data) {
      this.id = data.id;
      this.typeOfVacation = data.typeOfVacation;
      this.startOfVacationDate = new Date(data.startOfVacationDate);
      this.endOfVacationDate = new Date(data.endOfVacationDate);
      this.orderNumber = data.orderNumber;
      this.orderDate = new Date(data.orderDate);
      this.notes = data.notes;
      this.workingYearId = data.workingYearId;
    }
  }
}
